cli/tests/integrations/test_config.py

```diff
@@ -410,6 +416,9 @@ def test_prepend_url_validation(env):
 
 def test_fail_url_validation(env):
     _fail_url_validation('set', 'core.dcos_url', 'http://bad_domain/', env)
+    _fail_url_validation('set', 'core.dcos_url', 'http://@domain/', env)
+    _fail_url_validation('set', 'core.dcos_url', 'http://a.b:b.a@domain/', env)
+    _fail_url_validation('set', 'core.dcos_url', 'http://user:pass@/', env)
```

Comment:
we should test the actual user authentication exchange.  do we have a DCOS image with basic auth enabled?  If not, let's mock it.